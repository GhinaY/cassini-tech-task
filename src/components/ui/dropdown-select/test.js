import DropdownSelect from './index';
import { screen, render } from '@testing-library/react';

function renderComponent(props) {
  return(
    render(
      <DropdownSelect {...props}/>
    )
  );
};

describe('DropdownSelect', () => {
  const mockSetCurrentSelection = jest.fn();

  const mockOptions = ['option 1', 'option 2', 'option 3'];

  it('renders correctly with default optionl arguments', () => {
    const { asFragment } = renderComponent({ filterName: 'testDropdown', options: mockOptions, setCurrentSelection: mockSetCurrentSelection });
    
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('has an option for "All" when shouldShowOptionForAll is true', () => {
    renderComponent({ filterName: 'testDropdown', options: mockOptions, setCurrentSelection: mockSetCurrentSelection, shouldShowOptionForAll: true });

    expect(screen.getByText(/all/i)).toBeInTheDocument();
  });
});

// Definitely needs a lot more tests, including functional ones, 
// but in the interests of time I'll skip them for now.
