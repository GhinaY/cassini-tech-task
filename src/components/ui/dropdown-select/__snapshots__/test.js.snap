// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownSelect renders correctly with default optionl arguments 1`] = `
<DocumentFragment>
  <div
    class="DropdownSelectContainer"
  >
    <span>
      testDropdown:
    </span>
    <select
      class="FilterSelect"
      id="testDropdown"
      name="testdropdown"
    >
      <option
        value="option 1"
      >
        option 1
      </option>
      <option
        value="option 2"
      >
        option 2
      </option>
      <option
        value="option 3"
      >
        option 3
      </option>
    </select>
  </div>
</DocumentFragment>
`;
